extensions/amp-ad-network-doubleclick-impl/0.1/test/test-doubleclick-rtc.js

```diff
@@ -51,303 +46,70 @@ describes.realWin('DoubleClick Fast Fetch RTC', {amp: true}, env => {
   });
 
   afterEach(() => {
     sandbox.restore();
     impl = null;
-    xhrMock = null;
-    resetRtcStateForTesting();
-    const rtcConf = env.win.document.getElementById('amp-rtc');
-    env.win.document.head.removeChild(rtcConf);
-  });
-
-  function mockRtcExecution(rtcResponse, element, opt_textFunction) {
-    impl = new AmpAdNetworkDoubleclickImpl(element, env.win.document, env.win);
-    let textFunction = () => {
-      return Promise.resolve(JSON.stringify(rtcResponse));
-    };
-    textFunction = opt_textFunction || textFunction;
-    xhrMock.returns(
-        Promise.resolve({
-          redirected: false,
-          status: 200,
-          text: textFunction,
-        })
-    );
-    impl.populateAdUrlState();
-    return impl.executeRtc_(env.win.document);
-  }
-
-  function setRtcConfig(rtcConfigJson) {
-    const rtcConf = env.win.document.getElementById('amp-rtc');
-    rtcConf.innerText = JSON.stringify(rtcConfigJson);
-  }
-
-  it('should add just targeting to impl', () => {
-    const targeting = {'sport': 'baseball'};
-    const jsonTargeting = {
-      targeting,
-    };
-    return mockRtcExecution({
-      targeting,
-    }, element).then(() => {
-      expect(impl.jsonTargeting_).to.deep.equal(jsonTargeting);
-    });
   });
 
-  it('should add just categoryExclusions to impl', () => {
-    const categoryExclusions = {'sport': 'baseball'};
-    const jsonTargeting = {
-      categoryExclusions,
-    };
-    return mockRtcExecution({
-      categoryExclusions,
-    }, element).then(() => {
-      expect(impl.jsonTargeting_).to.deep.equal(jsonTargeting);
+  describe('#mergeRtcResponses_', () => {
+    function testMergeRtcResponses(
+        rtcResponseArray, expectedParams, expectedJsonTargeting) {
+      const rtcUrlParams = impl.mergeRtcResponses_(rtcResponseArray);
+      expect(rtcUrlParams).to.deep.equal(expectedParams);
+      expect(impl.jsonTargeting_).to.deep.equal(expectedJsonTargeting);
+    }
+    it('should properly merge RTC responses into jsonTargeting on impl', () => {
+      const rtcResponseArray = [
+        {response: {targeting: {'a': [1,2,3], 'b': {c: 'd'}}},
+          callout: 'www.exampleA.com', rtcTime: 100},
+        {response: {targeting: {'a': 'foo', 'b': {e: 'f'}}},
+          callout: 'www.exampleB.com', rtcTime: 500},
+        {response: {targeting: {'z': [{a: 'b'}, {c: 'd'}], 'b': {c: 'd'}}},
+          callout: 'www.exampleC.com', rtcTime: 100},
+      ];
+      const expectedParams = {
+        ati: '2,2,2',
+        artc: '100,500,100',
+        ard: 'www.exampleA.com,www.exampleB.com,www.exampleC.com',
+      };
+      const expectedJsonTargeting = {
+        targeting: {
+          'a': 'foo', 'b': {c: 'd', e: 'f'}, 'z': [{a: 'b'}, {c: 'd'}]},
+      };
+      testMergeRtcResponses(
+          rtcResponseArray, expectedParams, expectedJsonTargeting);
+    });
+
+    it('should only add params for callouts that were actually sent', () => {
+      const rtcResponseArray = [
```

Comment:
can we intermix these tests where we have some successful and some failure?

Reply:
the second one was intermixed, but I'll add more